### ISSUE TITLE
RSS feed view error

### ISSUE THREAD

**Issue by wouter76** [Xtraball/SiberianCMS#387]
As of the version 4.12.0 I detect an error in the rss. The list of rss items is loaded well but when trying to view one of the items the error occures. The error is at line 83 of the rss factory where the item is filtered from the entire collection.

So doing some test I detect the flowing is hapening:

1. when the collection is loaded first time by the `RssListController` the` data.collection` is set to the `$scope.collection` and the list view renderes without any problem.
2. next in this process the `$scope.collection` is copied into the  `Rss.collection` (at this point no undefined items in the array)
3. than the `data.cover` is added to the first position in the array, but without checking if the `data.cover` is empty. This causes the first item in the `Rss.collection` to be undefined .
4. when from the `RssViewController` then loading a feed the `Rss.getFeed(..)` is called
5. in the `Rss.getFeed` then a search in the `Rss.collection` is performed, but the first item in the `Rss.collection` array then is empty causing an error and ending up breaking the call leaving the user with an emptty (unrendered) view.

So my sugestion is to fix this bug in the rss.js file in the `RssListController`:
replace line 21 from

`Rss.collection.unshift(angular.copy(data.cover));`

to 
```
if (data.cover) {
    Rss.collection.unshift(angular.copy(data.cover));
}
```

**Comment by Vdvoem** [comment]
Hello!
Tell me in what file you made changes, the path to the file?
And tell me what errors are corrected?

**Comment by wouter76** [comment]
Hi,

I think I have to change all rss.js files in ..js/controllers/rss.js. .. meaning in browser and all ionic app dirs.

Cheers

**Comment by Vdvoem** [comment]
Hello and thank you for the info )
But the problem is not solved, in the function folder all errors.

**Comment by wouter76** [comment]
I see also the rss.bundle.js and rss.bundle.min.js have to change to get this working. 
I asume the dev team wil pick up this issue and publish a fix because this is in the core of the system.

**Comment by Vdvoem** [comment]
We have them of these problems have been written, a lot of them.

**Comment by dbyzero** [comment]
Thanks for the report, we will check ASAP.

**Comment by Slicit** [comment]
Fix is scheduled for 4.12.9 thanks.
